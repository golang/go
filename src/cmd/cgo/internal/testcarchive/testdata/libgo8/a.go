// Copyright 2021 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package main

import "C"

import (
	"os"
	"runtime"
	"sync/atomic"
)

var started int32

// Start a goroutine that loops forever.
func init() {
	runtime.GOMAXPROCS(1)
	go func() {
		for {
			atomic.StoreInt32(&started, 1)
		}
	}()
}

//export GoFunction8
func GoFunction8() {
	for atomic.LoadInt32(&started) == 0 {
		runtime.Gosched()
	}
	os.Exit(0)
}

func main() {
}
