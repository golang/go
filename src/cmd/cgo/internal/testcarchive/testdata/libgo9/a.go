// Copyright 2023 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package main

import "runtime"

// extern void callGoWithVariousStack(int);
import "C"

func main() {}

//export GoF
func GoF(p int32) {
	runtime.GC()
	if p != 0 {
		panic("panic")
	}
}

//export callGoWithVariousStackAndGoFrame
func callGoWithVariousStackAndGoFrame(p int32) {
	if p != 0 {
		defer func() {
			e := recover()
			if e == nil {
				panic("did not panic")
			}
			runtime.GC()
		}()
	}
	C.callGoWithVariousStack(C.int(p));
}
