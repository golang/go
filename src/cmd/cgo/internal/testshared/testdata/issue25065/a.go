// Copyright 2018 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

// Package issue25065 has a type with a method that is
//  1. referenced in a method expression
//  2. not called
//  3. not converted to an interface
//  4. is a value method but the reference is to the pointer method
//
// These cases avoid the call to makefuncsym from typecheckfunc, but we
// still need to call makefuncsym somehow or the symbol will not be defined.
package issue25065

type T int

func (t T) M() {}

func F() func(*T) {
	return (*T).M
}
