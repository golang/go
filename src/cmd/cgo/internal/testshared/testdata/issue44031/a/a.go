// Copyright 2021 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package a

type ATypeWithALoooooongName interface { // a long name, so the type descriptor symbol name is mangled
	M()
}
