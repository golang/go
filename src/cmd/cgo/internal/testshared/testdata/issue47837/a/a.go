// Copyright 2021 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package a

type A interface {
	M()
}

//go:noinline
func TheFuncWithArgA(a A) {
	a.M()
}

type ImplA struct{}

//go:noinline
func (A *ImplA) M() {}
