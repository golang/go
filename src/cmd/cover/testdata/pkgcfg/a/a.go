package a

type Atyp int

func (ap *Atyp) Set(q int) {
	*ap = Atyp(q)
}

func (ap Atyp) Get() int {
	inter := func(q Atyp) int {
		return int(q)
	}
	return inter(ap)
}

var afunc = func(x int) int {
	return x + 1
}
var Avar = afunc(42)

func A(x int) int {
	if x == 0 {
		return 22
	} else if x == 1 {
		return 33
	}
	return 44
}
