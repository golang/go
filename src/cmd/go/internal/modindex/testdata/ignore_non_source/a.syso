package ignore_non_source
