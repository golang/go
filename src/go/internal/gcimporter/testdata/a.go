// Copyright 2016 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

// Input for TestIssue13566

package a

import "encoding/json"

type A struct {
	a    *A
	json json.RawMessage
}
