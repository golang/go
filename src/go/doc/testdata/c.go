// Copyright 2012 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package c

import "a"

// ----------------------------------------------------------------------------
// Test that empty declarations don't cause problems

const ()

type ()

var ()

// ----------------------------------------------------------------------------
// Test that types with documentation on both, the Decl and the Spec node
// are handled correctly.

// A (should see this)
type A struct{}

// B (should see this)
type (
	B struct{}
)

type (
	// C (should see this)
	C struct{}
)

// D (should not see this)
type (
	// D (should see this)
	D struct{}
)

// E (should see this for E2 and E3)
type (
	// E1 (should see this)
	E1 struct{}
	E2 struct{}
	E3 struct{}
	// E4 (should see this)
	E4 struct{}
)

// ----------------------------------------------------------------------------
// Test that local and imported types are different when
// handling anonymous fields.

type T1 struct{}

func (t1 *T1) M() {}

// T2 must not show methods of local T1
type T2 struct {
	a.T1 // not the same as locally declared T1
}
