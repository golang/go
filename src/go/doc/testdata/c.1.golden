// 
PACKAGE c

IMPORTPATH
	testdata/c

IMPORTS
	a

FILENAMES
	testdata/c.go

TYPES
	// A (should see this) 
	type A struct{}

	// B (should see this) 
	type B struct{}

	// C (should see this) 
	type C struct{}

	// D (should see this) 
	type D struct{}

	// E1 (should see this) 
	type E1 struct{}

	// E (should see this for E2 and E3) 
	type E2 struct{}

	// E (should see this for E2 and E3) 
	type E3 struct{}

	// E4 (should see this) 
	type E4 struct{}

	// 
	type T1 struct{}

	// 
	func (t1 *T1) M()

	// T2 must not show methods of local T1 
	type T2 struct {
		a.T1	// not the same as locally declared T1
	}

